src/main/java/emu/grasscutter/game/player/Player.java

```diff
@@ -141,6 +142,8 @@ public class Player {
     @Getter private Map<Integer, ActiveCookCompoundData> activeCookCompounds;
     @Getter private Map<Integer, Integer> questGlobalVariables;
     @Getter private Map<Integer, Integer> openStates;
+    @Getter private Map<Integer, Set<Integer>> sceneTags;
+    @Getter private Set<Integer> visitedScenes;
```

Comment:
It might be better to use a Map<Integer,Map<Integer,Boolean>> to make it possible to identify added tags after version updates.
For visited scenes, maybe it could be `Map<Integer,Boolean>` instead and be used for unlocked scenes, that way, if a scene is in there, it got interacted with, and it can also support unlocking and locking scenes.

Reply:
It feels a little gross (I can't just convert the Set to a List and send it), but is needed for future expansion.